File: src/main/java/name/nkid00/rcutil/helper/GametimeHelper.java
package name.nkid00.rcutil.helper;

import net.minecraft.server.MinecraftServer;

public class GametimeHelper {
    private static long gametime;

    public static synchronized boolean isFrozen(MinecraftServer server) {
        return gametime == server.getOverworld().getTime() + 1;
    }

    public static synchronized void updateGametime(MinecraftServer server) {
        // START_SERVER_TICK is called before world time increases
        gametime = server.getOverworld().getTime() + 1;
    }

    public static synchronized long gametime() {
        return gametime;
    }
}

File: src/main/java/me/nk0/rcu/helper/GametimeHelper.kt
package me.nk0.rcu.helper

import net.minecraft.server.MinecraftServer

object GametimeHelper {
    private var gametime: Long = 0

    @JvmStatic
    @Synchronized
    fun isFrozen(server: MinecraftServer): Boolean {
        return gametime == server.overworld.time + 1
    }

    @JvmStatic
    @Synchronized
    fun updateGametime(server: MinecraftServer) {
        // START_SERVER_TICK is called before world time increases
        gametime = server.overworld.time + 1
    }

    @JvmStatic
    @Synchronized
    fun gametime(): Long {
        return gametime
    }
}
